## 第一周作业

#### 提交人学号：10235501454
#### 提交人姓名：陈子谦

#### 作业题目：
在0到1之间随机抽取两个数，求随机事件“两数之差大于1/3”的概率，分别采用频率方法和几何方法确定该概率。

#### 可参考的python代码

In [1]:
import random

def frequency_method(n):
    count = 0
    for _ in range(n):
        num1 = random.uniform(0, 1)
        num2 = random.uniform(0, 1)
        if abs(num1 - num2) > 1/3:
            count += 1
    pro = count / n
    return pro

n = 100000
pro_frequency = frequency_method(n)
print("频率方法得到的概率为:", pro_frequency)

频率方法得到的概率为: 0.44458


## 解答：
### 一、概率方法
    由频率方法的定义可知，题目所求的概率应为在n次重复试验中事件A “两数之差大于1/3” 出现的频数除以实验的总次数，于是我们使用Python代码来实现这一设想。
    首先代码中使用随机生成数random函数在(0,1)范围上随机生成两个数num1，num2。接着计算num1-num2的绝对值，若计算结果大于1/3，即满足事件A，就将计数器count的值增加1。
    对上述过程循环100000次使结果误差降低到可接受的范围
    最后将count除以实验次数(100000)即为频率方法所求出的概率：0.44212
### 二、几何方法(示意图在下方给出，若未显示请打开压缩包内的sample.jpg图片文件)
    由几何方法的定义可得，在直角坐标系下定义x为第一个随机数num1，y为第二个随机数num2。则当x∈(0,1) 且 y∈(0,1)时，其所围成的正方形区域为SΩ，事件A “两数之差大于1/3” 即为 |x-y|>1/3 
     ∵|x-y|>1/3
     => x-y>1/3 或 y-x>1/3
     => 图中蓝色阴影部分即为SA
     由几何方法求出的概率P = SA/SΩ = 4/9 ≈ 0.44
### 总结:
    最终我们可以发现几何方法所得概率为4/9，概率方法所得概率在0.4左右浮动，且当n取100000以上时稳定在0.44左右，故可得题目所求概率应约为0.44

![sample](./sample.jpg)